optaplanner-docs/src/main/asciidoc/Chapter-Partitioned_search/Chapter-Partitioned_search.adoc

```diff
@@ -8,27 +8,214 @@
-== Overview
+[[partitionedSearchAlgorithm]]
+== Algorithm Description
 
-For very big datasets, it is sometimes worthwhile to partition the datasets into smaller pieces.
+For big datasets, usually above 5000 planning entities,
+it is usually worthwhile to partition the datasets into smaller pieces and solve those separately.
+Partition Search is multi-threaded, leading to a performance boost due to more CPU utilization.
+Additionally, it finds an initial solution faster, even if it only uses only 1 CPU,
+because the search space sum of a partitioned Construction Heuristic is far less than its non-partitioned variant.
 
-However, **partitioning leads to suboptimal results**, even if the pieces are solved optimally:
+However, **partitioning does lead to suboptimal results**, even if the pieces are solved optimally, as shown below:
 
 image::Chapter-Partitioned_search/mapReduceIsTerribleForTsp.png[align="center"]
 
-A future version of Planner will have native support for several forms of partitioning.
-Meanwhile, you can implement it yourself as shown in the image above.
-Use an `Solver` to solve each piece.
+One way to deal with this is to run a non-partitioned Local Search after the Partitioned Search phase.
 
 [NOTE]
 ====
 Not all use cases can be partitioned.
-It only works on use cases for which the planning entities and value ranges can be divided into n pieces, such that none of the constraints cross piece boundaries.
+It only works on use cases for which the planning entities and value ranges can be split into n partition,
+such that none of the constraints cross the boundaries between partitions.
 ====
 
 
+[[partitionedSearchConfiguration]]
+== Configuration
+
+Simplest configuration:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+  </partitionedSearch>
+----
+
+There are several ways to <<partitioningASolution,partition a solution>>.
+
+Advanced configuration:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    ...
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+    <threadFactoryClass>...MyAppServerThreadFactory</threadFactoryClass>
+    <runnablePartThreadLimit>4</runnablePartThreadLimit>
+
+    <constructionHeuristic>...</constructionHeuristic>
+    <localSearch>...</localSearch>
+  </partitionedSearch>
+----
+
+The `threadFactoryClass` allows to plug in a custom `ThreadFactory` for environments
+that don't like arbitrary thread creation,
+such as most application servers, Android or Google App Engine.
+
+The `runnablePartThreadLimit` allows limiting CPU usage, see below.
+
+Just like a `<solver>` element, the `<partitionedSearch>` element can one or more <<solverPhase,phases>>.
+Each of those phases will be run on each partition.
```

Comment:
"Like the <solver> element, the <partitionedSearch> element can..." run? There is a word missing here.